src/main/java/hudson/remoting/URLish.java

```diff
@@ -16,20 +18,28 @@
 abstract class URLish {
     private URLish() {}
 
+    /**
+     * Converts URLish to the standard {@link URL} type.
+     * @return URL or {@code null} if the target destination is known to be non-existent.
```

Comment:
Why not throw, say, `FileNotFoundException` in such a case, so this can be `@Nonnull`?